cpp/util/SnowflakeCommon.cpp

```diff
@@ -10,10 +10,13 @@
 #include "snowflake/platform.h"
 #include "snowflake/Proxy.hpp"
 #include "../logger/SFLogger.hpp"
+#include <stdexcept>
 
 using namespace Snowflake;
 using namespace Snowflake::Client;
 
+static bool exception_on_memory_error = false;
```

Comment:
@sfc-gh-dprzybysz 
Sorry for the delay. @sfc-gh-ext-simba-jz is on vacation I'll take over.
Yes by default (for libsfc user other than odbc) it's false so no behavior change. For ODBC it would call sf_exception_on_memory_failure() to turn on exception on memory failures. ODBC is made with C++ and can handle exception well. The exception will cause ODBC API call to return SQL_ERROR instead of calling exit() to terminate customer application.